pkg/sql/logictest/testdata/logic_test/owner

```diff
@@ -255,3 +255,121 @@ user testuser
 
 statement error pq: role testuser cannot be dropped because some objects depend on it\nowner of database d\nowner of schema d.public\nowner of schema d.s\nowner of table d.public.t\nowner of table d.s.t\nowner of table test.public.t\nowner of type d.public._typ\nowner of type d.public.typ
 DROP ROLE testuser
+
+user root
+
+skipif config local-mixed-24.3
+let $testuser_job_id
+SELECT id FROM system.jobs WHERE owner = 'testuser' LIMIT 1
+
+skipif config local-mixed-24.3
+let $testuser2_job_id
+SELECT id FROM system.jobs WHERE owner = 'testuser2' LIMIT 1
+
+skipif config local-mixed-24.3
+let $node_job_id
+SELECT id FROM system.jobs WHERE owner = 'node' LIMIT 1
+
+skipif config local-mixed-24.3
+# Verify a simple transfer by an admin from testuser to testuser2 and back.
+statement ok
+ALTER JOB $testuser_job_id OWNER TO testuser2
+
+skipif config local-mixed-24.3
+query T
+SELECT owner FROM system.jobs WHERE id = $testuser_job_id
+----
+testuser2
+
+skipif config local-mixed-24.3
+statement ok
+ALTER JOB $testuser_job_id OWNER TO testuser
+
+skipif config local-mixed-24.3
+query T
+SELECT owner FROM system.jobs WHERE id = $testuser_job_id
+----
+testuser
+
+# Verify that ownership cannot transfer to or from 'node'.
+
+skipif config local-mixed-24.3
+statement error cannot transfer ownership
+ALTER JOB $node_job_id OWNER TO testuser
+
+skipif config local-mixed-24.3
+statement error cannot transfer ownership
+ALTER JOB $testuser_job_id OWNER TO node
+
+# Verify testuser2 can transfer ownership via role testuser to role testrole.
+user testuser2
+
+statement ok
+CREATE ROLE testrole
+
+statement ok
+CREATE ROLE otherrole
+
+statement ok
+CREATE USER testuser3
+
+statement ok
+GRANT testrole TO testuser3
+
+skipif config local-mixed-24.3
+statement ok
+ALTER JOB $testuser_job_id OWNER TO testrole
+
+skipif config local-mixed-24.3
+query T
+SELECT owner FROM system.jobs WHERE id = $testuser_job_id
+----
+testrole
+
+skipif config local-mixed-24.3
+# Verify testuser2 can transfer ownership to themselves and back to testuser.
+statement ok
+ALTER JOB $testuser_job_id OWNER TO testuser2
+
+skipif config local-mixed-24.3
+query T
+SELECT owner FROM system.jobs WHERE id = $testuser_job_id
+----
+testuser2
+
+skipif config local-mixed-24.3
+statement ok
+ALTER JOB $testuser_job_id OWNER TO testuser
```

Comment:
wait, why can we transfer ownership to `testuser`? The job is currently owned by `testuser2` which is apart of the `testrole`, but `testuser` is not a member of this role.

Reply:
this is where reading a diff is harder than reading the actual file: in the existing test setup, before this added section, on line 152 we do `GRANT testuser TO testuser2;` so testuser2 is a member of the testuser role already.

Reply:
hrm. this test would probably be cleaner in its own file, but i also understand that you're on a crusade to minimize test server inits.

Reply:
I donno, the file is all about ownership both via direct and via roles, and is only a couple hundred lines, so this feels like where this belongs doesn't it? also, yeah, not worth an extra test server startup.